### ISSUE TITLE
Direct assignment of page.onWhatever functions is broken.

### ISSUE THREAD

**Issue by lukesneeringer** [foray1010/node-phantom-promise#1]
First of all, thanks for `node-phantom-promise`. I like the notion of returning thenables rather than passing callbacks around as it makes for cleaner code.

Unfortunately, it looks like your wrapper breaks one important thing, which is that setting functions to `page.onConsoleMessage`, `page.onCallback`, etc., no longer works.

Assume a PhantomJS page under the variable `page`, opened using the standard pattern:

``` javascript
phantom.create({path: require('phantomjs').path}).then((engine) => {
  engine.createPage().then((page) => {
    ...
  });
});
```

I should be able to set certain properties to `page` directly:

``` javascript
page.onConsoleMessage = (msg, lineno, src) => {
  console.log(msg);
};
```

And then it should work: 

``` javascript
// This should log out 'foo' and 'bar', possibly in the reverse order.
page.evaluate(function() {
  console.log('foo');
  return 'bar';
}).then((result) => console.log(result));
```

It doesn't, though. The `onConsoleMessage` function never fires. **It does in `node-phantom-simple`, though, so something with the wrapper is breaking it.**

You can get around this, kind of, by using `setFn`, which still works under the wrapper:

``` javascript
page.setFn('onConsoleMessage', function(msg) { console.log(msg) }).then(() => {
  page.evaluate(function() {
    console.log('foo');
    return 'bar';
  }).then((result) => console.log(result));
});
```

But this is an insufficient workaround because `setFn` (in `node-phantom-simple`, not in your code) actually converts the function to a string and then runs it through `eval`, meaning that all function context is lost (this makes baby moogle cry, kupo!).

Anyway, it would be awesome if direct assignment of `page.onWhatever = function () { ... }` worked again. I _think_ it needs to be some sort of exemption in `promisifyAll` but I wasn't certain enough to put together a pull request.

**Comment by foray1010** [comment]
thanks for the report! Overlooked this one before.

The reason is phantomjs looks into the original `page` to get `onWhatever` callback
I have come up with some solutions:
1. just use the original object and make all promisified function name with an `Async` in the end, such as `openAsync`
2. Make a wrapper: page.on('ConsoleMessage', callback) for setting the function, page.on('ConsoleMessage') for getting the function

I don't like the first one, I will see if I can figure out another way without changing the behaviour, if not, the change will be in `1.1.0`

**Comment by foray1010** [comment]
Fixed in https://github.com/foray1010/node-phantom-promise/commit/2c05e8be3d450a4bbd71a2a39a1ba70291b15f9c and released `1.0.2`

Found some smart ticks using Object.defineProperties to link the prop between promisified `page` and original `page`
